connector-email/src/main/java/com/redhat/cloud/notifications/connector/email/processors/recipients/RecipientsFilter.java

```diff
@@ -0,0 +1,84 @@
+package com.redhat.cloud.notifications.connector.email.processors.recipients;
+
+import com.redhat.cloud.notifications.connector.email.constants.ExchangeProperty;
+import com.redhat.cloud.notifications.connector.email.model.User;
+import com.redhat.cloud.notifications.connector.email.model.settings.RecipientSettings;
+import org.apache.camel.Exchange;
+import org.apache.camel.Processor;
+
+import javax.enterprise.context.ApplicationScoped;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+@ApplicationScoped
+public class RecipientsFilter implements Processor {
+    /**
+     * Filters the fetched users according to the criteria specified in the
+     * recipient settings.
+     * @param exchange the exchange of the pipeline.
+     */
+    @Override
+    public void process(final Exchange exchange) {
+        // Fetch the required data to filter the users.
+        final boolean isOptIn = exchange.getProperty(ExchangeProperty.IS_OPT_IN, Boolean.class);
+        final RecipientSettings recipientSettings = exchange.getProperty(ExchangeProperty.CURRENT_RECIPIENT_SETTINGS, RecipientSettings.class);
+        final Set<String> subscribers = (Set<String>) exchange.getProperty(ExchangeProperty.SUBSCRIBERS, Set.class);
+        Set<User> users = (Set<User>) exchange.getProperty(ExchangeProperty.USERS, Set.class);
+
+        // If the request contains a list of users, then the recipients from RBAC who are not included in
+        // the request users list are filtered out.
+        // Otherwise, the full list of recipients from RBAC will be processed by the next step.
+        if (recipientSettings.getUsers().size() > 0) {
+            users = filterUsers(users, recipientSettings.getUsers());
+        }
+
+        // If the user preferences should be ignored, the recipients from RBAC (possibly filtered by the
+        // previous step) is returned without filtering out the users who didn't subscribe to the event type.
+        if (recipientSettings.isIgnoreUserPreferences()) {
+            users = Set.copyOf(users);
+        } else {
+            if (isOptIn) {
+                // Otherwise, the recipients from RBAC who didn't subscribe to the event type are filtered out.
+                users = filterUsers(users, subscribers);
+            } else {
+                users = filterUnsubscribedUsers(users, subscribers);
+            }
```

Comment:
This logic was dedicated to the Drawer in the old code. We could make things simpler in this connector and ignore the opt in/out logic entirely (which includes not passing the boolean in `EmailNotification`).

Reply:
We removed this, and therefore it doesn't make sense to pass the subscribed users to the email connector anymore, right?

Reply:
Your question made me realize something's missing in the current code.

Reply:
Oh... right. Reverted!